### PR TITLE
Round off x axis max timestamp to T - 1 min and 0th second

### PR DESCRIPTION
**Pull request by YounixM**
### Summary
Round off x axis max timestamp to T - 1 min and 0th second

#### Related Issues / PR's
https://github.com/SigNoz/signoz/pull/4144#issuecomment-1857543595

Before:

290780397-36bd3180-8461-4f44-b548-e36582892cff

Screenshot 2023-12-15 at 3 44 21 PM

After:
Screenshot 2023-12-15 163448

image

### CODE REVIEW

**Review by srikanthccv — APPROVED** [review]
Looks good now

Screenshot 2023-12-15 at 5 13 50 PM

File: frontend/src/lib/uPlotLib/utils/getXAxisScale.ts [hunk 1]
```diff
@@ -36,5 +36,18 @@ export const getXAxisScale = (
 		maxTime = fallbackMax;
 	}
 
+	// As API response is adjusted to return values for  T - 1 min (end timestamp) due to which graph would not have the current timestamp data, we see the empty space in the graph for smaller timeframe. To accommodate this, we will be plotting the graph from (startTime  ->  endTime - 1 min).
```
Comment (palashgdev) [inline frontend/src/lib/uPlotLib/utils/getXAxisScale.ts:39]: good to add a note why this is required

**Review by palashgdev — APPROVED** [review]
LGTM verified ✅

File: frontend/src/lib/uPlotLib/utils/getXAxisScale.ts (hunk 1 above)
Comment (srikanthccv) [reply frontend/src/lib/uPlotLib/utils/getXAxisScale.ts:39]: _As API response is adjusted to return values for  T - 1 min (end timestamp) due to which graph would not have the current timestamp data, we see the empty space in the graph for smaller timeframe._ This part explains why it is required

**Pull request merged**